src/common/src/types/with_data_type.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::rc::Rc;
+use std::sync::Arc;
+
+use bytes::Bytes;
+
+use super::{
+    DataType, Date, Decimal, Fields, Int256, Interval, JsonbRef, JsonbVal, StructType, Time,
+    Timestamp, Timestamptz, F32, F64,
+};
+
+pub trait WithDataType {
```

Comment:
This is a little hard to understand why it’s needed. Add some comments?

Reply:
I can’t find similar thing that can be used, but I guess we may use it in somewhere. I’ll review it later.

Currently it’s just a macro helper.